### PR TITLE
Depends: bump to qt 5.5

### PR DESCRIPTION
**Pull request by theuni**
This is a continuation of #6314. Thanks to @welshjf for getting this rolling.

It's not as big/complicated as it looks at first glance, there are a few changes that add to the size:
- Split sources are now used rather than the huge complete tarball. 225M -> 75M. Thanks again to @welshjf for that.
- config options are now mostly one-per-line to make diffs easier to read

It's pretty straightforward otherwise.

Worth mentioning: I had to add 2 hacks for building with mingw. We need to make sure that all works well on win 8.1 and win10, as it looks like that's the path taken for the new (patched) code.

### CODE REVIEW

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -1,53 +1,132 @@
...
-$(package)_config_opts += -skip qtsvg -skip qtwebkit -skip qtwebkit-examples -skip qtserialport
-$(package)_config_opts += -skip qtdeclarative -skip qtmultimedia -skip qtimageformats -skip qtx11extras
-$(package)_config_opts += -skip qtlocation -skip qtsensors -skip qtquick1 -skip qtxmlpatterns
-$(package)_config_opts += -skip qtquickcontrols -skip qtactiveqt -skip qtconnectivity -skip qtmacextras
-$(package)_config_opts += -skip qtwinextras -skip qtxmlpatterns -skip qtscript -skip qtdoc
-
-$(package)_config_opts += -prefix $(host_prefix) -bindir $(build_prefix)/bin
-$(package)_config_opts += -no-c++11 -openssl-linked  -v -static -silent -pkg-config
-$(package)_config_opts += -qt-libpng -qt-libjpeg -qt-zlib -qt-pcre
+$(package)_config_opts += -opensource -confirm-license
```
Comment (droark) [inline depends/packages/qt.mk]: Just to be nit-picky, I assume this line needs to be split in two if it's going to be consistent with the style being used.

**Comment by laanwj** [comment]
Thanks!
Needs to go into 0.11.1 as well for #6426.

**Comment by jonasschnelli** [comment]
Great! Like to see this go in.
Linux build over gitian somehow doesn't work atm: https://builds.jonasschnelli.ch/pulls/6471/build-linux.log

Windows and OSX build is okay.
Now testing the Windows and Osx executables.

**Comment by jonasschnelli** [comment]
Testes on Win 8 and OSX 10.7 as well as on 10.10. Works fine.
On OSX there is a tiny issue (coming in with Qt5.5 because they have fixed HiDPIness of QLabel) with the (new) bitcoin logo in the left top corner of the about screen (will PR).

bildschirmfoto 2015-07-24 um 11 40 52

bildschirmfoto 2015-07-24 um 11 40 14

bildschirmfoto 2015-07-24 um 11 38 16



bildschirmfoto 2015-07-24 um 11 46 20
bildschirmfoto 2015-07-24 um 11 46 01

bildschirmfoto 2015-07-24 um 11 34 27

**Comment by laanwj** [comment]
Get the following error on `make` inside depends, first line:
```
/bin/sh: 1: test: qtbase-opensource-src-5.5.0.tar.gz: unexpected operator
```
It does apprear to continue after it.

**Comment by theuni** [comment]
@jonasschnelli wow, thanks for the extensive tests! I just pushed a new commit that should fix the Linux build.

@laanwj That can be safely ignored for now, it's a trivial syntax thing. I'd like to fix it, but the fix requires touching funcs.mk, which will force a rebuild of all packages. I'll take care of it next time something in there needs to be changed.

**Comment by jonasschnelli** [comment]
Tested ACK.
Gitian build works now also for linux. Tested on all three platforms.

bildschirmfoto 2015-07-25 um 00 06 52

**Comment by laanwj** [comment]
Merged via f6850d58f7dd87de44dd25be0fc1067625fa8d3a - removed the duplicate -no-reduce-relocations

**Comment by laanwj** [comment]
Backported to 0.11 as 92401c2d9051e5d07b9f5608604925575ebaffef

**Comment by welshjf** [comment]
I've now tested this (as the 0.11 backport), and while my original 5.4 bump did not fix the systray icon on KDE 5, this one finally does. Yay.

**Comment by theuni** [comment]
@welshjf great news, thanks for testing.

**Pull request closed without merging**